src/pages/moderators-page/components/main-content/index.tsx

```diff
@@ -1,30 +1,136 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import { useDispatch } from "react-redux";
 
-import { useGetReviewsModerationQuery } from "~/redux/reviews-moderation/reviews-moderation-api";
+import { Modal, SearchBar, Spinner } from "~/common/components";
+import { ScreenBreakpoints } from "~/common/constants";
+import { ButtonGroupData, SpinnerVariant } from "~/common/enums";
+import { useGetScreenWidth } from "~/common/hooks";
+import { DropdownOption, GetModerationReviewsRequest } from "~/common/types";
+import { NotFound } from "~/pages/home-page/components/main-content/components/search-block/components";
+import useReviewModerationApi from "~/redux/reviews-moderation/hookUseReviewModerationApi";
 import { setRewiews } from "~/redux/reviews-moderation/reviews-moderation-slice";
 
+import {
+	ModeratorsReviewFilterSection,
+	ReviewModeratorsCard,
+} from "./components/index";
 import styles from "./styles.module.scss";
 
+const INDEX_ZERO = 0;
+
 const MainModeratorsContent: React.FC = () => {
-	const { data: moderatorsReviews } = useGetReviewsModerationQuery();
+	const [searchTerm, setSearchTerm] = useState<string>("");
+	const [filterByStatus, setFilterByStatus] =
+		useState<DropdownOption["value"]>();
+	const [sortByPeriod, setSortByPeriod] = useState<DropdownOption["value"]>();
+	const [filterByType, setFilterByType] =
+		useState<keyof typeof ButtonGroupData>("Компанії");
+	const [isOpenSerchFiltersModal, setIsOpenSerchFiltersModal] = useState(false);
+	const screenWidth = useGetScreenWidth();
+
+	const handleSetSearchTerm = (term: string) => {
+		setSearchTerm(term);
+	};
+
+	// const handleOpenSearchFilter = () => {
+	// 	setIsOpenSerchFiltersModal(true);
+	// };
```

Comment:
delete comment

Reply:
i live for the future refact? this is need later